src/main/java/edu/ucsb/cs156/happiercows/controllers/StudentsController.java

```diff
@@ -0,0 +1,121 @@
+package edu.ucsb.cs156.happiercows.controllers;
+import edu.ucsb.cs156.happiercows.entities.Students;
+import edu.ucsb.cs156.happiercows.repositories.StudentsRepository;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+import io.swagger.v3.oas.annotations.Operation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import edu.ucsb.cs156.happiercows.errors.EntityNotFoundException;
+import io.swagger.v3.oas.annotations.Parameter;
+import org.springframework.web.bind.annotation.RequestParam;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import java.time.LocalDateTime;
+import javax.validation.Valid;
+
+@Tag(name = "Students")
+@RequestMapping("/api/Students")
+@RestController
+@Slf4j
+public class StudentsController extends ApiController {
+
+    @Autowired
+    StudentsRepository StudentsRepository;
+
+    /**
+     * List all students
+     * 
+     * @return an iterable of Students
+     */
+    @Operation(summary= "List all students")
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
```suggestion
    @PreAuthorize("hasRole('ROLE_ADMIN')")
```

We should not be releasing lists of students to anyone other than the Admins.